internal/resources/grafana/resource_sso_settings_test.go

```diff
@@ -0,0 +1,292 @@
+
+	provider := "github"
+
+	api := testutils.Provider.Meta().(*common.Client).GrafanaOAPI.WithOrgID(1)
+	defaultSettings, err := api.SsoSettings.GetProviderSettings(provider)
+	if err != nil {
+		t.Fatalf("failed to fetch the default settings for provider %s: %v", provider, err)
+	}
+
+	resource.ParallelTest(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		CheckDestroy:      checkSsoSettingsReset(api, provider, defaultSettings.Payload),
+		Steps: []resource.TestStep{
+			{
+				Config: testutils.TestAccExample(t, "resources/grafana_sso_settings/resource.tf"),
+				Check: resource.ComposeTestCheckFunc(
+					ssoSettingsCheckExists.exists("grafana_sso_settings.github_sso_settings", &settings),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "provider_name", provider),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "oauth2_settings.#", "1"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "oauth2_settings.0.client_id", "github_client_id"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "oauth2_settings.0.client_secret", "github_client_secret"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "oauth2_settings.0.team_ids", "12,50,123"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.github_sso_settings", "oauth2_settings.0.allowed_organizations", "organization1,organization2"),
+				),
+			},
+			{
+				ResourceName:            "grafana_sso_settings.github_sso_settings",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"oauth2_settings.0.client_secret"},
+			},
+		},
+	})
+}
+
+// multiple SSO settings resources having the same provider are sharing a single entry in the DB
+func TestSSOSettings_twoResourcesForOneProvider(t *testing.T) {
+	testutils.CheckOSSTestsEnabled(t, ">=10.4.0")
+
+	provider := "azuread"
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: testutils.ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testConfigWithTwoResourcesForOneProvider,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("grafana_sso_settings.first_sso_settings", "provider_name", provider),
+					resource.TestCheckResourceAttr("grafana_sso_settings.first_sso_settings", "oauth2_settings.#", "1"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.first_sso_settings", "oauth2_settings.0.client_id", "first_gitlab_client_id"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "provider_name", provider),
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "oauth2_settings.#", "1"),
+					resource.TestCheckResourceAttr("grafana_sso_settings.second_sso_settings", "oauth2_settings.0.client_id", "second_gitlab_client_id"),
+				),
+				ExpectNonEmptyPlan: true,
```

Comment:
There's a drift?
Usually `ExpectNonEmptyPlan` is an error condition. Seems weird to check for attributes at the same time

Reply:
Yeah, that's a weird test/scenario. The idea is that the key for a `grafana_sso_settings` resource is the `provider_name` field. So if we have 2 terraform resources with different settings but with the same provider name then there will be a drift in the terraform plan because the 2 resources share the same DB record but have 2 entries in the terraform state.
This test passes when I run it locally with the required API. I'll try to make it more clear.

Reply:
ehh, not sure it's worth testing this tbh. I'd remove it, personally. It's not testing the behaviour of the resource, it's testing Terraform itself